autocards.py

```diff
@@ -1,63 +1,285 @@
 from pipelines import qg_pipeline
-from transformers import pipeline
+
+from tqdm import tqdm
+from pathlib import Path
+import pandas as pd
+import time
+import signal
+import pdb
+import re
+import os
+from contextlib import suppress
+
+import requests
+import PyPDF2
+import wikipedia
+from wikipedia.exceptions import PageError
 from bs4 import BeautifulSoup
-import csv
+from pprint import pprint
+
+# otherwise csv and json outputs contain a warning string
+os.environ["TOKENIZERS_PARALLELISM"] = "true"
 
 
 class Autocards:
     def __init__(self):
-        self.qg = qg_pipeline('question-generation', model='valhalla/t5-base-qg-hl', ans_model='valhalla/t5-small-qa-qg-hl')
+        print("Loading backend...")
+        self.qg = qg_pipeline('question-generation',
+                              model='valhalla/t5-base-qg-hl',
+                              ans_model='valhalla/t5-small-qa-qg-hl')
         self.qa_pairs = []
+        global n, cur_n
```

Comment:
Yeah I decided to do it like that because I was not sure of when the line "n= len(self.qa_pairs)" woud be executed otherwise. For example it has to be reset when `clear_qa` is run.

Looking back it's actually a way more terrible idea that I thought because I can imagine **most** users loading autocards already having set a "n" variable somewhere...

Unfortunately I don't feel comfortable correcting this myself as I'm unfamiliar with classes. It's actually my first ever class... Would you be interested in explaining me the best route or doing it yourself? I will surely have to ankify that...

Reply:
to be more clear: I don't feel up to solving when and how to set n and cur_n. I can take care of the issue of organizing qa_pairs differently later on myself